docs/connecting_to_remote_db_aws_aurora.md

```diff
@@ -0,0 +1,112 @@
+# Connecting to a remote Klaxon database in an AWS Aurora cluster
+
+This documentation assumes several things:
+- The app is running locally in a Docker container
+- You have already created an Aurora RDS cluster where the Klaxon database will be housed. That cluster should be configured in a way that is compatible with PostgreSQL 14. For our purposes at the Post, we used `database_engine_version: 14.4` and `instance_size: db.t4g.medium`. [Here is a guide](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html#Concepts.DBInstanceClass.Support) to help you select a compatible instance size. We deploy our AWS resources in infrastructure-specific repos, however resources [can also be created in the AWS console](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.CreateInstance.html).
+- You have master credentials to access the Postgres database.
+
+## Create the database and appuser
+
+First, we need to create a database called `klaxon` on the new cluster and add a user role that will be used by our app. You will need the master credentials and the endpoint for the cluster, which can be [found in the AWS console](https://docs.aws.amazon.com/documentdb/latest/developerguide/db-cluster-endpoints-find.html). Open a psql shell on the remote cluster like:
+```
+psql --host=<CLUSTER ENDPOINT> --username=<MASTER USERNAME> --password
+```
+
+This will prompt you for the master password. Then, you will be in the psql shell where you can run:
+```
+CREATE DATABASE klaxon;
+CREATE ROLE <app username> WITH LOGIN PASSWORD '<password>';
+GRANT ALL PRIVILEGES ON DATABASE klaxon TO <app username>;
+```
+
+Leave this shell open, we'll come back to it at the end!
+
+## Create a new environment
+
+Open a new terminal window for the following steps.
+
+Unlike the test, development, production environments (for automated tests, local development, and live production, respectively), we use local, dev, and prod. Regardless of what convention your team uses, we want to create an `.env` file that will contain the credentials for this remote db (and eventually, the remote app). Create that file with whatever name you choose. We use:
```

Comment:
❓ this is something I think I'd like to chat more about @reviewers because I'm not sure the best way to match the existing klaxon environments with our typical 🌊 flow 🌊